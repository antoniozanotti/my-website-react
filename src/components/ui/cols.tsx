import { ReactNode } from "react";

export interface ColsProps extends React.ComponentProps<"div"> {
  children?: ReactNode;
}

export default function Cols({ children, ...props }: ColsProps) {
  let divClasses = "flex gap-x-4";
  props.className = props.className
    ? divClasses + " " + props.className
    : divClasses;
  return <div {...props}>{children}</div>;
}
