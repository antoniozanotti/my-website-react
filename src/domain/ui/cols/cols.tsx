export function Cols({
  children,
  ...props
}: React.ComponentProps<"div">) {
  const baseStyles = "flex gap-x-4";
  props.className = props.className
    ? `${baseStyles} ${props.className}`
    : baseStyles;

  return <div {...props}>{children}</div>;
}
