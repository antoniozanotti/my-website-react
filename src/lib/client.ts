import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
