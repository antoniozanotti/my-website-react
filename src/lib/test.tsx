import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
