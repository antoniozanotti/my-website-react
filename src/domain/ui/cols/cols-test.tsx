import { test, expect } from '@playwright/experimental-ct-react';
import { Cols } from "./cols";

test.describe("Cols", () => {
  test("should render div", async ({ mount }) => {
    const component = await mount(<Cols />);
    await expect(component.locator(":scope:is(div)")).toHaveCount(1);
  });

  test("should accept className", async ({ mount }) => {
    const component = await mount(<Cols className='test' />);
    await expect(component).toHaveClass(/test$/);
  });

  test("should render children", async ({ mount }) => {
    const component = await mount(<Cols>Test</Cols>);
    await expect(component).toContainText("Test");
  });
});
