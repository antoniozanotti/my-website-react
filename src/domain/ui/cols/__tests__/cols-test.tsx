import { screen, render, firstComponent } from "@/lib/test";
import { Cols } from "../cols";

describe("Cols", () => {
  it("should render div", async () => {
    const { container } = render(<Cols />);
    const component = firstComponent(screen);

    expect(component).toBeVisible();
    expect(container).toMatchSnapshot();
  });

  it("should accept className", async () => {
    const { container } = render(<Cols className='test' />);
    const component = firstComponent(screen);

    expect(component).toHaveClass(/test$/);
    expect(container).toMatchSnapshot();
  });

  it("should render children", async () => {
    const { container } = render(<Cols>Test</Cols>);
    const component = firstComponent(screen);
    
    expect(component).toHaveTextContent("Test");
    expect(container).toMatchSnapshot();
  });
});
