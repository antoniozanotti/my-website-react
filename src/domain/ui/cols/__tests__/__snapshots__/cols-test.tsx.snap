// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cols should accept className 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="flex gap-x-4 test"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Cols should render children 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="flex gap-x-4"
        >
          Test
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cols should render div 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="flex gap-x-4"
        />
      </div>
    </div>
  </div>
</div>
`;
